src/RuleSet/RuleSets.php

```diff
@@ -89,7 +92,7 @@ public static function registerRuleSet(string $name, string $class): bool
 
         $set = new $class();
 
-        if (!$set instanceof RuleSetDescriptionInterface) {
+        if (!\in_array(RuleSetDescriptionInterface::class, class_implements($set), true)) {
```

Comment:
```php
        $set = new $class(); // ← TO REMOVE

        if (!\in_array(RuleSetDescriptionInterface::class, class_implements($class), true)) {
```

As I said, instantiation is not needed, we can use `$class` directly.

Reply:
Oh, sorry, I misunderstood you – or wasn't paying enough attention.